app/src/main/java/io/apicurio/registry/storage/impl/kafkasql/KafkaSqlRegistryStorage.java

```diff
@@ -634,6 +635,30 @@ public void importData(EntityInputStream entities, boolean preserveGlobalId, boo
         });
     }
 
+    /**
+     * @see io.apicurio.registry.storage.RegistryStorage#importData(io.apicurio.registry.storage.impexp.EntityInputStream,
+     *      boolean, boolean)
+     */
+    @Override
+    public void upgradeData(EntityInputStream entities, boolean preserveGlobalId, boolean preserveContentId)
+            throws RegistryStorageException {
+        DataImporter dataImporter = new SqlDataUpgrader(log, utils, this, preserveGlobalId,
+                preserveContentId);
+        dataImporter.importData(entities, () -> {
+            // Because importing just pushes a bunch of Kafka messages, we may need to
+            // wait for a few seconds before we send the reset messages. Due to partitioning,
+            // we can't guarantee ordering of these next two messages, and we NEED them to
+            // be consumed after all the import messages.
+            // TODO We can wait until the last message is read (a specific one),
+            // or create a new message type for this purpose (a sync message).
+            try {
+                Thread.sleep(2000);
+            } catch (Exception e) {
+                // Noop
+            }
```

Comment:
I know this was probably copy/pasted from somewhere, but I think it's no longer true with the new KafkaSQL implementation.  In the new impl, all messages should be on the same partition.  So there should no longer need to be a 2s wait to ensure ordering.

Reply:
Yup, it's copy pasted, the existing importData method still does this. I'll leave it as-is now and re-visit both in the future. I'll add a TODO.